pkg/controller/cronjob/cronjob_controller2.go

```diff
@@ -0,0 +1,580 @@
+		if err := ratelimiter.RegisterMetricAndTrackRateLimiterUsage("cronjob_controller", kubeClient.CoreV1().RESTClient().GetRateLimiter()); err != nil {
+			return nil, err
+		}
+	}
+
+	jm := &Controller2{
+		queue:    workqueue.NewNamedDelayingQueue("cronjob2"),
+		recorder: eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "cronjob-controller"}),
+
+		jobControl:     realJobControl{KubeClient: kubeClient},
+		cronJobControl: &realCJControl{KubeClient: kubeClient},
+
+		jobLister:     jobInformer.Lister(),
+		cronJobLister: cronJobsInformer.Lister(),
+
+		jobListerSynced:     jobInformer.Informer().HasSynced,
+		cronJobListerSynced: cronJobsInformer.Informer().HasSynced,
+		now:                 time.Now,
+	}
+
+	jobInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    jm.addJob,
+		UpdateFunc: jm.updateJob,
+		DeleteFunc: jm.deleteJob,
+	})
+
+	cronJobsInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			jm.enqueueController(obj)
+		},
+		UpdateFunc: jm.updateCronJob,
+		DeleteFunc: func(obj interface{}) {
+			jm.enqueueController(obj)
+		},
+	})
+
+	return jm, nil
+}
+
+// Run starts the main goroutine responsible for watching and syncing jobs.
+func (jm *Controller2) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	klog.Infof("Starting cronjob controller")
+
+	if !cache.WaitForNamedCacheSync("job", stopCh, jm.jobListerSynced, jm.cronJobListerSynced) {
+		return
+	}
+
+	for i := 0; i < workers; i++ {
+		go wait.Until(jm.worker, time.Second, stopCh)
+	}
+
+	<-stopCh
+	klog.Infof("Shutting down CronJob Manager")
+}
+
+func (jm *Controller2) worker() {
+	for jm.processNextWorkItem() {
+	}
+}
+
+func (jm *Controller2) processNextWorkItem() bool {
+	key, quit := jm.queue.Get()
+	if quit {
+		return false
+	}
+	defer jm.queue.Done(key)
+	err, requeueAfter := jm.sync(key.(string))
+	switch {
+	case err != nil:
+		utilruntime.HandleError(fmt.Errorf("Error syncing CronJobController %v, requeuing: %v", key.(string), err))
+		jm.queue.Add(key)
+	case requeueAfter != nil:
+		jm.queue.AddAfter(key, *requeueAfter)
+	default:
+		jm.queue.Done(key)
+	}
+	return true
+}
+
+func (jm *Controller2) sync(cronJobKey string) (error, *time.Duration) {
+	ns, name, err := cache.SplitMetaNamespaceKey(cronJobKey)
+	if err != nil {
+		return err, nil
+	}
+
+	cronJob, err := jm.cronJobLister.CronJobs(ns).Get(name)
+	switch {
+	case errors.IsNotFound(err):
+		// may be cronjob is deleted, dont need to requeue this key
+		nameForLog := fmt.Sprintf("%s/%s", ns, name)
+		klog.V(4).Infof("cronjob %s not found, may be it is deleted: %#v", nameForLog, err)
+		return nil, nil
+	case err != nil:
+		// for other transient apiserver error requeue with exponential backoff
+		return err, nil
+	}
+
+	// TODO: @alpatel limit the amount of data we scrape here
+	jobList, err := jm.jobLister.Jobs(ns).List(labels.Everything())
```

Comment:
We've talked about having a label selector, put that in the comment, so whoever gets to it, knows what we want.

Reply:
found an easy way to achieve this added it now, PTAL